.github/workflows/reusable-on-code-change.yml

```diff
@@ -0,0 +1,198 @@
+name: Reusable Workflow (On Code Change)
+on:
+  workflow_call:
+    inputs:
+        static_analysis_instance_type:
+            type: string
+            default: "ubuntu-latest"
+        build_instance_type:
+            type: string
+            default: "ubuntu-latest"
+        sonar_analysis_instance_type:
+            type: string
+            default: "ubuntu-latest"
+        integration_tests_instance_type:
+            type: string
+            default: "self-hosted"
+        tests_standalone:
+            type: boolean
+            default: false
+        tests_cluster:
+            type: boolean
+            default: false
+        node_version:
+            type: number
+            default: 18
+        mvn_settings_filepath:
+            type: string
+            default: ".github/maven.settings.xml"
+        static_analysis_auditci_level:
+            type: string
+            default: "moderate"
+        module_id:
```

Comment:
module_id is an example of a variable that should be global even though it might, right now, only be used by one job.